opty/utils.py

```diff
@@ -229,6 +230,140 @@ def parse_free(free, n, q, N):
     return free_states, free_specified, free_constants
 
 
+def create_objective_function(
+    objective, state_symbols, input_symbols, unknown_symbols,
+    N, node_time_interval=1.0, integration_method="backward euler"):
+    """Creates function to evaluate the objective and objective gradient.
+
+    Parameters
+    ----------
+    objective : sympy.Expr
+        The objective function to be minimized, which is a function of the
+        states and inputs.
+    state_symbols : iterable of symbols
+        The state variables.
+    input_symbols : iterable of symbols
+        The input variables.
+    unknown_symbols : iterable of symbols
+        The unknown parameters.
+    N : int
+        Number of collocation nodes, i.e. the number of time steps.
+    node_time_interval : float
+        The value of the time interval. The default is 1.0, as this term only
+        appears in the objective function as a scaling factor.
+    integration_method : str, optional
+        The method used to integrate the system. The default is "backward
+        euler".
+
+    """
+    def lambdify_function(expr, multiplication_array, take_sum):
+        if take_sum:
+            def integration_function(x):
+                return node_time_interval * np.sum(x * multiplication_array)
+        else:
+            def integration_function(x):
+                return node_time_interval * x * multiplication_array
```

Comment:
Did you try an see if lambidy(module=scipy) works to convert these integrals to the proper scipy function?

Reply:
ah `NotImplementedError: Only definite integrals are supported`

Reply:
I did not check, but thanks for checking. I have never experimented with what the scipy printer does. I should do that some time.

Reply:
I think it works with definite integrals. I'm wondering if we need these special cases in this function it integrals already work in lambdify.

Reply:
We would still like to specify the integration method it uses.

Reply:
When using `scipy` with properly set limits, then it uses [quad](https://docs.scipy.org/doc/scipy/reference/generated/scipy.integrate.quad.html)

Reply:
It should be possible to support integrals with limits. It can be done by updating the multiplication array. Only bit difficult part is of course that you may have different limits for different integrals in the same function.